.github/copilot-instructions.md

```diff
@@ -0,0 +1,154 @@
+├── dDatabase/            # Database abstraction layer
+├── dGame/                # Core game logic, components, behaviors
+├── dScripts/             # Game scripts (NPCs, quests, etc.)
+├── dNet/                 # Network utilities
+├── dPhysics/             # Physics integration
+├── tests/                # Unit tests (GoogleTest)
+├── migrations/           # Database schema migrations
+├── thirdparty/           # External dependencies
+├── build.sh              # Main build script
+├── CMakeVariables.txt    # Build configuration variables
+└── CMakePresets.json     # CMake preset configurations
+```
+
+### Key Files to Know
+- **CMakeVariables.txt**: Build configuration (testing enabled, MariaDB jobs, etc.)
+- **build/sharedconfig.ini**: Database connection, client location, server settings
+- **build/masterconfig.ini**: Master server port and startup configuration
+- **CONTRIBUTING.md**: Code style guidelines and commit message format
+- **docs/Commands.md**: Complete list of in-game server commands
+
+### Build Configuration
+Located in `CMakeVariables.txt`:
+- `ENABLE_TESTING=1` - Unit tests enabled (keep enabled)
+- `MARIADB_CONNECTOR_COMPILE_JOBS=1` - Parallel compilation jobs for MariaDB connector
+- `CDCLIENT_CACHE_ALL=0` - Database caching strategy
+
+### Common Commands Reference
+```bash
+# Build from clean state
+rm -rf build && ./build.sh -j2
+
+# Run specific test
+cd build && ctest -R "TestName" --output-on-failure
+
+# Check which servers were built
+cd build && ls -la *Server
+
+# View build configuration
+cat CMakeVariables.txt
+
+# Check git submodules status
+git submodule status
+```
+
+### Important Notes
+- **Client files are NOT included** - this is only the server emulator
+- **Database can use SQLite or MariaDB** - MariaDB recommended for development
+- **Multi-server architecture** requires all 4 servers to run a complete setup
+- **Network ports**: Auth (1001), Chat (2005), Master (2000), World (3000+)
+- **Development uses Debug builds**, production uses Release builds
+- **GMT level 0** = normal player, **GMT level 8-9** = admin privileges
+
+## Troubleshooting
+- **"Asset bundle not found"**: Expected without LEGO Universe client files
+- **"Submodule errors"**: Run `git submodule update --init --recursive`
+- **"CMake version errors"**: Requires CMake 3.25-3.31
+- **"MariaDB connection errors"**: Check database setup and sharedconfig.ini
+- **"Permission denied on port"**: Run `sudo setcap 'cap_net_bind_service=+ep' AuthServer` for ports <1024
+
+### CI Information
+- **GitHub Actions** runs builds on Windows, Ubuntu, and macOS  
+- **Build matrix** tests multiple configurations via CMake presets
+- **All tests must pass** for CI to succeed
+- **Build artifacts** are automatically generated and uploaded
+
+**Remember: This is a complex game server requiring LEGO Universe client files for full functionality, but the server code can be built, tested, and partially validated without them.**
```

Comment:
And the server has the capability to mock everything that's needed to test without the client since cdlient can be mocked and the database can be mocked as well.